draft-ietf-privacypass-architecture.md

```diff
@@ -141,123 +164,193 @@ framework.
+  to this capability for the purposes of being ruled out as a bot or otherwise
+  automated Client.
+- Valid account. Clients that possess valid application layer account
+  identifiers, e.g., because they are paid subscribers for some serice, can
+  attest to this fact.
+- Trusted device. Some Clients run on trusted hardware that are capable of
+  producing device-level attestation statements.
+
+Each of these attestation types have different security properties. For
+example, attesting to having a valid account is different from attesting to be
+running on trusted hardware. In general, Mediators should accept a limited form
+of attestation formats.
+
+Each attestation format also has an impact on the overall system privacy. For
+example, the number of users in possession of a single class of trusted device
+might be lesser than the number of users that can solve CAPTCHAs. Similarly,
+requiring a conjunction of attestation types could decrease the overall
+anonymity set size. For example, the number of Clients that have solved a
+CAPTCHA in the past day, have a valid account, and are running on a trusted
+device is lesser thant he number of Clients that have solved a CAPTCHA in the
+past day. Mediators should not admit attestation types that result in small
+anonymity sets.
+
+#### Accounting
+
+Another important role of the Mediator is accounting. This is necessary for
+some issunace protocols that enforce rate limits. For example, the Issuer may
+want to limit the number of tokens issued to a single Client over the course
+of a day. If the Issuer does not learn information about the Client, then the
+Issuer cannot enforce this limit on a per-Client basis. The Issuer could,
+however, enforce global rate limits, but these can be abused by malicious
+Clients at the determent of honest Clients. Thus, the task of accounting
+falls on the Mediator, since it is entrusted with some amount of
+privacy-sensitive Client information.
+
+### Issuer Role
+
+Issuers MUST be uniquely identifiable by all Clients with a consistent
+identifier. In a web context, this identifier might be the Issuer host name.
+As discussed later in {{privacy}}, ecosystems that admit a large number of
+Issuers can lead to privacy concerns for the Clients in the ecosystem.
+Therefore, in practice, the number of Issuers should be bounded. The actual
+Issuers can be replaced with different Issuers as long as the total never
+exceeds these bounds. Moreover, Issuer replacements also have an effect on
+client anonymity that is similar to when a key rotation occurs. See {{privacy}}
+for more details about maintaining privacy with multiple servers.
+
+#### Key Management
+
+To facilitate issuance, the Issuer MUST hold a Privacy Pass key pair at any
+given time. The server public key MUST be made available to all Clients in
+such a way that key rotations and other updates are publicly visible. The
+server MAY also require additional state for ensuring this.  The key material
```

Comment:
unclear: for ensuring what?

Reply:
I'm not sure -- this is leftover text, so I'll just remove it.